### PR TITLE
php 8.0 support

### PR DESCRIPTION
**Pull request by mollux**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Minimal changes to support PHP 8.0 for Mautic 4 (therefore it's based on 4.3).

This is the successor of #9969 (and makes #11048 obsolete).

To support PHP 8.1,  more (breaking) changes are needed, as dependencies (of dependencies) only support PHP 8.1 in a version with BC breaking changes.

#### Steps to test this PR:

* check the outcome of the test
* go through Mautic with error reporting on and reporting all errors

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11217"></a>

### CODE REVIEW

File: .github/ci-files/composer.json
Comment (RCheesley) [inline .github/ci-files/composer.json]: ❓ I think it would be better if we still have to maintain this fork, that it were on a Mautic org repo rather than an individuals. Just thinking out loud, but then we can ensure that it is subject to our own security processes/branch protections and the like.

**Review by RCheesley — CHANGES_REQUESTED** [review]
I think if we are still having to use that fork we should bring it under the Mautic org so we can control the security side of things. Other than that thanks for doing this work @mollux - have not tested it as yet, this was just a first pass over the proposed changes.

File: .github/ci-files/composer.json
Comment (mollux) [reply .github/ci-files/composer.json]: @RCheesley I created a PR to be able to get it in Dennis' fork (see https://github.com/dennisameling/FOSOAuthServerBundle/pulls), but indeed it's safer to move it to the mautic org.

Comment (RCheesley) [reply .github/ci-files/composer.json]: I think that would be preferable nowadays if we are still having to support the fork - from the supply chain perspective I think we should be a bit tighter on this kind of thing just as a belt and braces approach!

Comment (mollux) [reply .github/ci-files/composer.json]: agreed! can you create a repo for this and give me access? I can move the code

Comment (RCheesley) [reply .github/ci-files/composer.json]: https://github.com/mautic/FOSOAuthServerBundle voila (I just forked it direct from the main repo)

Comment (mollux) [reply .github/ci-files/composer.json]: @RCheesley  I switched to that repo

**Comment by mollux** [comment]
@RCheesley 

the CS Fixer and PHPSTAN report don't work, as the name changed from `PHPSTAN` | `CS Fixer`  to  `PHPSTAN - 7.4` | `CS Fixer - 7.4`.
This seems to be configured somewhere in a setting of the project that I can't access.

This is something we will have to figure out if it can be configured per branch or not

File: .github/ci-files/composer.json
Comment (RCheesley) [reply .github/ci-files/composer.json]: OK great - will just need to bring over any of our changes to that repo as well then ;)

Comment (mollux) [reply .github/ci-files/composer.json]: I added them in https://github.com/mautic/mautic/pull/11217/commits/607876fa220d5e49c1fa1d24871357276ab79f1f

Comment (RCheesley) [reply .github/ci-files/composer.json]: Sorry, I mean bring over our changes to https://github.com/mautic/FOSOAuthServerBundle from Dennis' fork, as the fork I made was from the original repo so it won't have our changes applied, I am guessing. Or if that is a problem maybe we can just ask Dennis if we can move the forked repo into our org.

Comment (mollux) [reply .github/ci-files/composer.json]: Those changes are present, see https://github.com/mautic/FOSOAuthServerBundle/tree/doctrine-fix
I pushed the exact same branch from Denis' fork + my php 8 compatibility change, so we're good in this case.

Comment (RCheesley) [reply .github/ci-files/composer.json]: Ahhh perfect, sorry I was not looking at the branch! 🤦

**Review by escopecz — DISMISSED** [review]
I see no problem with the code changes. Nicely done, Mattias! 👍

**Review by volha-pivavarchyk — DISMISSED** [review]
All changes look good. Tests works well as before

**Comment by mollux** [comment]
@escopecz how do we tackle https://github.com/mautic/mautic/pull/11217#issuecomment-1147753062 ? I have no access, and don't know if that label can be configured per branch in github

**Comment by escopecz** [comment]
I was thinking that the CS Fixer and PHPSTAN stages are there only because the base branch have them. I wanted to try to merge it and see what happens. What do you think?

**Comment by escopecz** [comment]
I'm trying to merge this to see if the old CS Fixer and PHPSTAN jobs disappear. I think they are present only because the target branch has them.

**Comment by escopecz** [comment]
Sadly, my theory that the orphan jobs will disappear after merge was false. I merged fresh 4.3 branch into https://github.com/mautic/mautic/pull/11256 and the jobs are hanging there too.

**Comment by escopecz** [comment]
So it was hidden in the branch settings. All good now 🎉  https://github.community/t/removed-workflow-showing-up-as-check-for-pr/16393

**Pull request merged**
